### PR TITLE
Update v7 projects to the new csproj

### PR DESCRIPTION
**Pull request by timbussmann**
In order to be able to target .net core in a next step, we need to switch to the new csproj style.

ping @bording @ramonsmits

### CODE REVIEW

**Comment by ramonsmits** [comment]
@timbussmann I'll review and test Tomorrow morning.

**Comment by ramonsmits** [comment]
Getting compiler errors. Currently updating VS to latest version.

**Comment by ramonsmits** [comment]
Running vs2017 15.4.2 resolved the compiler errors.

File: src/PerformanceTests/Persistence.V7.Azure.V8/Persistence.V7.Azure_v8.csproj [hunk 1]
```diff
@@ -1,106 +1,21 @@
...
-    <ProjectGuid>{4B4045B2-1B52-44B7-9CC2-42C761B56C9B}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <TargetFramework>net461</TargetFramework>
     <RootNamespace>Persistence.V7.Azure.V8</RootNamespace>
     <AssemblyName>Persistence.V7.Azure.V8</AssemblyName>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <OutputPath>..\bin\$(Configuration)\Persistence.V7.Azure_v8\</OutputPath>
+    <DefineConstants>$(DefineConstants);Version7</DefineConstants>
```
Comment (ramonsmits) [inline src/PerformanceTests/Persistence.V7.Azure.V8/Persistence.V7.Azure_v8.csproj]: Nice!

File: src/PerformanceTests/Transport.V7.RabbitMQ.V5/RabbitMQProfile.cs [hunk 2]
```diff
@@ -10,14 +10,15 @@ class RabbitMQProfile : IProfile, INeedPermutation
...
-        var cs = ConfigurationHelper.GetConnectionString("RabbitMQ");
+        var cs = ConfigurationHelper.GetConnectionString("RabbitMQ2");
         var builder = new DbConnectionStringBuilder { ConnectionString = cs };
         if (builder.Remove("prefetchcount")) NServiceBus.Logging.LogManager.GetLogger(nameof(RabbitMQProfile)).Warn("Removed 'prefetchcount' value from connection string.");
 
         var transport = endpointConfiguration
             .UseTransport<RabbitMQTransport>()
             .UseConventionalRoutingTopology()
-            .PrefetchMultiplier(Permutation.PrefetchMultiplier);
+            .PrefetchMultiplier(Permutation.PrefetchMultiplier)
```
Comment (ramonsmits) [inline src/PerformanceTests/Transport.V7.RabbitMQ.V5/RabbitMQProfile.cs]: As you've hardcoded this option, might we want to backport this to the previous major versions?

File: src/PerformanceTests/Transport.V7.RabbitMQ.V5/RabbitMQProfile.cs [hunk 3]
```diff
@@ -10,14 +10,15 @@ class RabbitMQProfile : IProfile, INeedPermutation
 
     public void Configure(EndpointConfiguration endpointConfiguration)
     {
-        var cs = ConfigurationHelper.GetConnectionString("RabbitMQ");
+        var cs = ConfigurationHelper.GetConnectionString("RabbitMQ2");
```
Comment (ramonsmits) [inline src/PerformanceTests/Transport.V7.RabbitMQ.V5/RabbitMQProfile.cs]: I would prefer just having a single connection string, but if we have multiple let us create a better connection string name then nr. 2

File: src/PerformanceTests/Persistence.V6.NHibernate.V7/NHibernateProfile.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ public void Configure(EndpointConfiguration cfg)
     void ISetup.Setup()
     {
         var cs = ConfigurationHelper.GetConnectionString(Permutation.Persister.ToString());
-        var sql = Assembly.GetExecutingAssembly().GetManifestResourceText("Persistence.V6.NHibernate.init.sql");
+        var sql = Assembly.GetExecutingAssembly().GetManifestResourceText("Persistence.V7.NHibernate.init.sql");
```
Comment (ramonsmits) [inline src/PerformanceTests/Persistence.V6.NHibernate.V7/NHibernateProfile.cs]: This is an error I think?

File: src/PerformanceTests/Transport.V7.MQMQ_v1/MsmqProfile.cs [hunk 5]
```diff
@@ -14,7 +14,7 @@ class MsmqProfile : IProfile, INeedPermutation
     public void Configure(EndpointConfiguration endpointConfiguration)
     {
         var transport = endpointConfiguration.UseTransport<MsmqTransport>();
-        transport.ConnectionString(ConfigurationHelper.GetConnectionString("MSMQ"));
+        transport.DisableDeadLetterQueueing();
```
Comment (ramonsmits) [inline src/PerformanceTests/Transport.V7.MQMQ_v1/MsmqProfile.cs]: Same as the RabbitMQ one, if this is hardcoded. Shall we make it fixed in previous majors too?

**Comment by ramonsmits** [comment]
Getting the following runtime error for SQS which I cannot seem to get resolved: 
```
Amazon.Runtime.AmazonClientException: No RegionEndpoint or ServiceURL configured
   at Amazon.Runtime.ClientConfig.Validate()
   at Amazon.S3.AmazonS3Config.Validate()
   at Amazon.Runtime.AmazonServiceClient..ctor(AWSCredentials credentials, ClientConfig config)
   at NServiceBus.AmazonSQS.ConnectionConfiguration.<>c.<get_S3ClientFactory>b__6_0()
   at NServiceBus.TransportInfrastructure..ctor(ReadOnlySettings settings)
   at NServiceBus.SqsTransport.Initialize(SettingsHolder settings, String connectionString)
```

**Comment by ramonsmits** [comment]
Azure Service Bus works, Azure Storage Queues does not.

**Comment by ramonsmits** [comment]
I tried explicit references for ASQ but that still results in assembly load errors:

```xml
    <PackageReference Include="Microsoft.Data.Edm" Version="5.8.3" />
    <PackageReference Include="Microsoft.Data.OData" Version="5.8.3" />
    <PackageReference Include="Microsoft.Data.Services.Client" Version="5.8.3" />
    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
    <PackageReference Include="WindowsAzure.Storage" Version="8.4.0" />
```

Error:

```
2017-11-01 16:19:38.3169|DEBUG| 1|FirstChanceException|Could not load file or assembly 'Microsoft.Data.Edm, Version=5.8.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Data.Edm, Version=5.8.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'Microsoft.Data.Edm, Version=5.8.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
   at System.Reflection.RuntimeAssembly.nLoadFile(String path, Evidence evidence)
   at System.Reflection.Assembly.LoadFile(String path)
```

**Comment by ramonsmits** [comment]
SQS works on master, I think it is an issue with the NServiceBus.AmazonSQS package or one of its dependencies. NServiceBus.AmazonSQS-4.0.0

@Particular/aws-maintainers I really do not have a clue what might be wrong here. Any idea what could be wrong?

**Comment by ramonsmits** [comment]
Explicit packages for ASQ do seem to have worked after a complete cleanup and a rebuild however, there seems to be a bug in the ASQ transport: https://github.com/Particular/NServiceBus.AzureStorageQueues/issues/287

**Comment by bording** [comment]
> Explicit packages for ASQ do seem to have worked after a complete cleanup and a rebuild however

This shouldn't be required. If it is, then it we need to fix the ASQ package. It might mean raising to a newer minimum WindowsAzure.Storage if there is indeed a problem here.

**Comment by danielmarbach** [comment]
@ramonsmits is the ENV variable `AWS_DEFAULT_REGION` set? https://docs.particular.net/transports/upgrades/amazonsqs-3to4

**Comment by bording** [comment]
IF this PR is about switching to the new project system, why are there so many changes in here that don't appear to be related to that? It's making this very hard to review.

**Comment by ramonsmits** [comment]
Did a rebase of the branch, excluding irrelevant changes @timbussmann @bording to everything atleast running. Other changes to fix the breaking changes in core 7 beta will be put in a different branch.

File: src/PerformanceTests/Persistence.V6.NHibernate.V7/NHibernateProfile.cs (hunk 4 above)
Comment (timbussmann) [reply src/PerformanceTests/Persistence.V6.NHibernate.V7/NHibernateProfile.cs]: yeah seems so

File: src/PerformanceTests/Transport.V7.RabbitMQ.V5/RabbitMQProfile.cs (hunk 3 above)
Comment (timbussmann) [reply src/PerformanceTests/Transport.V7.RabbitMQ.V5/RabbitMQProfile.cs]: happy with a better name, but let's do that in a different PR

File: src/PerformanceTests/Transport.V7.RabbitMQ.V5/RabbitMQProfile.cs (hunk 2 above)
Comment (timbussmann) [reply src/PerformanceTests/Transport.V7.RabbitMQ.V5/RabbitMQProfile.cs]: I don't mind. Feel free to "backport" this

**Comment by timbussmann** [comment]
going ahead and merge this  to unblock other PRs. With ramons changes I think this PR again became a little bit cleaner although the changeset remains quite large :( I think it's easier at that point to fix missed details in upcoming PRs.

**Pull request merged**